autogpt/memory/vectara.py

```diff
@@ -0,0 +1,291 @@
+"""Implementation of memory backend using Vectara (https://vectara.com)"""
+import hashlib
+import json
+import logging
+import time
+
+import requests
+from colorama import Fore, Style
+
+from autogpt.config import Config
+from autogpt.logs import logger
+from autogpt.memory.base import MemoryProviderSingleton
+
+
+class JWTTokenProvider:
+    """Provides OAuth2 JWT tokens for authorization."""
+
+    def __init__(self, customer_id, app_id, app_secret):
+        self._app_id = app_id
+        self._app_secret = app_secret
+        self._token_endpoint = f"https://vectara-prod-{customer_id}.auth.us-west-2.amazoncognito.com/oauth2/token"
```

Comment:
how likely is this to change?

Reply:
Highly unlikely. All our customers use the token endpoint this way.
If we ever change this, we will have to coordinate with all existing customers.